app/models/folio/item.rb

```diff
@@ -110,10 +109,14 @@ def status_class
     end
 
     def status_text
-      if !circulates?
+      if !requestable? && temporary_location.present?
+        temporary_location.discovery_display_name
+      elsif !circulates?
```

Comment:
I'm not sure "Unavailable" is quite the right text here (the item status "Aged to lost" is the most likely item status that results in it not being requestable, which is.. worse), but 🤷‍♂️ 

```suggestion
      return temporary_location.discovery_display_name || 'Unavailable' if !requestable?

      if !circulates?
```